<!-- 
emba - EMBEDDED LINUX ANALYZER

Copyright 2020-2021 Siemens AG
Copyright 2020-2021 Siemens Energy AG

emba comes with ABSOLUTELY NO WARRANTY. This is free software, and you are
welcome to redistribute it under the terms of the GNU General Public License.
See LICENSE file for usage of this software.

emba is licensed under GPLv3

Author(s): Michael Messner, Pascal Eckmann, Stefan Haboeck
-->

<!-- Used icons: https://github.com/CoreyGinnivan/system-uicons -->
 
<!DOCTYPE html>
<html lang="en">
<head>
  <title>EMBA firmware report</title>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="stylesheet" href="./../style/style.css" type="text/css">
  <link rel="icon" type="image/png" href="./../style/favicon.png">
</head>
<body>

<div id="sidenav" class="side">
  <div class="logo">
    <a class="inherit" href="./../index.html"><img class="inherit" src="./../style/emba.svg" alt="emba logo"></a>
  </div>
  <div id="nav">
      <a class="back" href="./../index.html">&laquo; Back to main</a>
<a class="modul" href="./../s21_python_check.html" title="./../s21_python_check.html" >&laquo; Back to s21_python_check</a>
      <!-- navigation start -->
      <!-- navigation end -->
      <input id="expand" class="expand_njs hidden" type="button" value="More results:" onclick="this.value=this.value=='+ Show more results'?'- Hide more results':'+ Show more results';">
      <div class="expand_area">
        <!-- etc start -->
        <!-- etc end -->
      </div>
  </div>
  <a href="TIMELINK"><div id="buttonTimeInvisible" class="rectButton"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" width="40" height="40" style="-ms-transform: rotate(360deg); -webkit-transform: rotate(360deg); transform: rotate(360deg);" preserveAspectRatio="xMidYMid meet" viewBox="0 0 21 21"><g fill="none" fill-rule="evenodd" stroke="white" stroke-linecap="round" stroke-linejoin="round" transform="matrix(-1 0 0 1 19 2)"><circle cx="8.5" cy="8.5" r="8"/><path d="M8.5 5.5v4H5"/></g></svg></div></a>
  <div id="buttonSizer" class="rectButton buttonClose" onclick="buttonNav()"></div>
</div>
<div id="main" class="main">
  <!-- content start -->
<pre>************* Module datetime</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1125:0: R1707: Disallow trailing comma tuple (trailing-comma-tuple)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1:0: C0302: Too many lines in module (2518/1000) (too-many-lines)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:11:0: C0103: Argument name "x" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:11:0: C0103: Argument name "y" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:31:0: C0103: Constant name "dbm" doesn't conform to UPPER_CASE naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:43:4: C0103: Variable name "y" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:61:4: W0621: Redefining name 'dim' from outer scope (line 32) (redefined-outer-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:83:0: C0103: Argument name "n" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:118:4: C0103: Variable name "n4" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:122:4: C0103: Variable name "n1" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:151:0: C0103: Argument name "y" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:151:0: C0103: Argument name "m" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:151:0: C0103: Argument name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:151:0: C0103: Argument name "hh" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:151:0: C0103: Argument name "mm" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:151:0: C0103: Argument name "ss" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:151:0: R0913: Too many arguments (7/5) (too-many-arguments)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:156:0: C0103: Argument name "hh" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:156:0: C0103: Argument name "mm" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:156:0: C0103: Argument name "ss" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:156:0: C0103: Argument name "us" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:173:8: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:178:4: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:185:8: C0103: Variable name "hh" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:185:12: C0103: Variable name "mm" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:186:8: C0103: Variable name "mm" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:186:12: C0103: Variable name "ss" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:187:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:189:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:192:16: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:196:19: W0622: Redefining built-in 'object' (redefined-builtin)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:196:27: W0622: Redefining built-in 'format' (redefined-builtin)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:196:0: R0914: Too many local variables (18/15) (too-many-locals)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:200:4: C0103: Variable name "Zreplace" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:205:7: C0103: Variable name "n" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:207:8: C0103: Variable name "ch" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:211:16: C0103: Variable name "ch" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:228:32: C0103: Variable name "h" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:229:32: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:230:32: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:231:32: C0103: Variable name "u" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:206:4: R1702: Too many nested blocks (8/5) (too-many-nested-blocks)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:242:24: C0103: Variable name "Zreplace" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:244:28: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:247:32: C0103: Variable name "Zreplace" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:206:4: R1702: Too many nested blocks (7/5) (too-many-nested-blocks)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:196:0: R0912: Too many branches (20/12) (too-many-branches)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:196:0: R0915: Too many statements (55/50) (too-many-statements)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:300:8: R1720: Unnecessary "else" after "raise" (no-else-raise)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:345:12: C0103: Variable name "td" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:402:8: C0415: Import outside toplevel (warnings) (import-outside-toplevel)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:419:4: W0621: Redefining name 'dim' from outer scope (line 32) (redefined-outer-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:441:0: C0103: Argument name "tz" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:445:0: C0103: Argument name "x" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:445:0: C0103: Argument name "y" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:449:0: C0103: Argument name "a" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:449:0: C0103: Argument name "b" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:457:4: C0103: Variable name "q" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:457:7: C0103: Variable name "r" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:461:4: C0103: Variable name "r" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:464:8: C0103: Variable name "q" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:469:0: C0103: Class name "timedelta" doesn't conform to PascalCase naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:488:4: R0913: Too many arguments (8/5) (too-many-arguments)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:488:4: R0914: Too many local variables (17/15) (too-many-locals)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:502:8: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:502:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:502:16: C0103: Variable name "us" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:515:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:517:12: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:520:12: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:541:8: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:542:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:555:12: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:556:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:561:12: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:562:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:570:17: C0103: Variable name "us" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:571:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:572:14: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:573:8: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:488:4: R0915: Too many statements (68/50) (too-many-statements)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:604:8: C0103: Variable name "mm" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:604:12: C0103: Variable name "ss" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:605:8: C0103: Variable name "hh" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:605:12: C0103: Variable name "mm" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:606:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:608:12: C0103: Argument name "n" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:610:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:612:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:672:8: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:686:12: C0103: Variable name "a" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:686:15: C0103: Variable name "b" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:696:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:714:12: C0103: Variable name "a" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:714:15: C0103: Variable name "b" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:705:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:719:12: C0103: Variable name "r" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:725:12: C0103: Variable name "q" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:725:15: C0103: Variable name "r" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:733:8: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:739:8: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:745:8: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:751:8: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:757:8: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:764:38: W0212: Access to a protected member _getstate of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:789:0: C0103: Class name "date" doesn't conform to PascalCase naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:835:20: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:840:12: W0212: Access to a protected member __setstate of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:854:4: C0103: Argument name "t" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:856:8: C0103: Variable name "y" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:856:11: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:856:14: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:856:17: C0103: Variable name "hh" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:856:21: C0103: Variable name "mm" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:856:25: C0103: Variable name "ss" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:856:17: W0612: Unused variable 'hh' (unused-variable)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:856:21: W0612: Unused variable 'mm' (unused-variable)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:856:25: W0612: Unused variable 'ss' (unused-variable)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:856:29: W0612: Unused variable 'weekday' (unused-variable)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:856:38: W0612: Unused variable 'jday' (unused-variable)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:856:44: W0612: Unused variable 'dst' (unused-variable)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:862:8: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:866:4: C0103: Argument name "n" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:872:8: C0103: Variable name "y" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:872:11: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:872:14: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:885:12: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1049:8: C0103: Variable name "y" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1049:11: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1049:14: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1050:8: C0103: Variable name "y2" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1050:12: C0103: Variable name "m2" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1050:16: C0103: Variable name "d2" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1050:21: W0212: Access to a protected member _year of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1050:34: W0212: Access to a protected member _month of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1050:48: W0212: Access to a protected member _day of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1064:12: C0103: Variable name "o" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1134:0: C0103: Class name "_date_class" doesn't conform to PascalCase naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1141:0: C0103: Class name "tzinfo" doesn't conform to PascalCase naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1148:4: C0103: Argument name "dt" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1152:4: C0103: Argument name "dt" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1156:4: C0103: Argument name "dt" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1164:4: C0103: Argument name "dt" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1196:19: E1102: getinitargs is not callable (not-callable)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1201:20: E1102: getstate is not callable (not-callable)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1204:8: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1209:0: C0103: Class name "_tzinfo_class" doesn't conform to PascalCase naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1211:0: C0103: Class name "time" doesn't conform to PascalCase naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1236:64: W0621: Redefining name 'tzinfo' from outer scope (line 1141) (redefined-outer-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1236:4: R0913: Too many arguments (6/5) (too-many-arguments)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1254:20: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1259:12: W0212: Access to a protected member __setstate of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1310:8: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1316:8: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1322:8: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1328:8: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1334:8: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1342:15: W0212: Access to a protected member _tzinfo of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1355:25: W0212: Access to a protected member _hour of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1355:38: W0212: Access to a protected member _minute of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1355:53: W0212: Access to a protected member _second of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1356:25: W0212: Access to a protected member _microsecond of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1358:12: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1363:17: W0212: Access to a protected member _hour of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1363:36: W0212: Access to a protected member _minute of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1365:29: W0212: Access to a protected member _second of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1365:44: W0212: Access to a protected member _microsecond of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1371:16: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1373:16: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1378:16: C0103: Variable name "h" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1378:19: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1381:16: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1398:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1400:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1402:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1403:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1408:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1411:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1423:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1425:8: C0103: Variable name "tz" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1427:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1441:12: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1502:16: W0621: Redefining name 'tzinfo' from outer scope (line 1141) (redefined-outer-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1501:4: R0913: Too many arguments (6/5) (too-many-arguments)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1523:8: C0103: Variable name "h" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1525:12: C0103: Variable name "h" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1528:8: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1533:33: W0621: Redefining name 'tzinfo' from outer scope (line 1141) (redefined-outer-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1536:8: C0103: Variable name "h" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1552:0: C0103: Class name "_time_class" doesn't conform to PascalCase naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1558:0: C0103: Class name "datetime" doesn't conform to PascalCase naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1558:0: R0902: Too many instance attributes (10/7) (too-many-instance-attributes)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1567:31: W0621: Redefining name 'tzinfo' from outer scope (line 1141) (redefined-outer-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1566:4: R0913: Too many arguments (9/5) (too-many-arguments)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1576:20: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1581:12: W0212: Access to a protected member __setstate of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1632:4: C0103: Argument name "t" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1632:4: C0103: Argument name "tz" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1632:4: R0914: Too many local variables (21/15) (too-many-locals)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1638:8: C0103: Variable name "us" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1641:12: C0103: Variable name "us" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1644:12: C0103: Variable name "us" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1647:8: C0103: Variable name "y" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1647:11: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1647:14: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1647:17: C0103: Variable name "hh" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1647:21: C0103: Variable name "mm" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1647:25: C0103: Variable name "ss" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1648:8: C0103: Variable name "ss" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1663:12: C0103: Variable name "y" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1663:15: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1663:18: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1663:21: C0103: Variable name "hh" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1663:25: C0103: Variable name "mm" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1663:29: C0103: Variable name "ss" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1667:16: C0103: Variable name "y" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1667:19: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1667:22: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1667:25: C0103: Variable name "hh" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1667:29: C0103: Variable name "mm" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1667:33: C0103: Variable name "ss" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1670:20: W0212: Access to a protected member _fold of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1647:29: W0612: Unused variable 'weekday' (unused-variable)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1647:38: W0612: Unused variable 'jday' (unused-variable)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1647:44: W0612: Unused variable 'dst' (unused-variable)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1676:4: W0221: Parameters differ from overridden 'fromtimestamp' method (arguments-differ)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1686:4: C0103: Argument name "t" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1691:4: C0103: Argument name "tz" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1693:8: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1699:8: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1703:21: W0621: Redefining name 'date' from outer scope (line 789) (redefined-outer-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1703:27: W0621: Redefining name 'time' from outer scope (line 1211) (redefined-outer-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1703:33: W0621: Redefining name 'tzinfo' from outer scope (line 1141) (redefined-outer-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1728:12: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1734:16: W0707: Consider explicitly re-raising using the 'from' keyword (raise-missing-from)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1757:8: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1758:8: C0103: Argument name "u" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1759:12: C0103: Variable name "y" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1759:15: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1759:18: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1759:21: C0103: Variable name "hh" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1759:25: C0103: Variable name "mm" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1759:29: C0103: Variable name "ss" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1763:8: C0103: Variable name "a" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1764:8: C0103: Variable name "u1" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1765:8: C0103: Variable name "t1" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1770:12: C0103: Variable name "u2" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1771:12: C0103: Variable name "b" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1775:12: C0103: Variable name "b" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1777:8: C0103: Variable name "u2" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1778:8: C0103: Variable name "t2" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1790:8: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1791:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1801:8: C0103: Variable name "y" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1801:11: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1801:14: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1802:8: C0103: Variable name "hh" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1802:12: C0103: Variable name "mm" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1802:16: C0103: Variable name "ss" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1819:60: W0621: Redefining name 'tzinfo' from outer scope (line 1141) (redefined-outer-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1818:4: W0221: Parameters differ from overridden 'replace' method (arguments-differ)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1818:4: R0913: Too many arguments (9/5) (too-many-arguments)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1845:12: C0103: Variable name "ts" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1847:12: C0103: Variable name "ts" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1849:8: W0612: Unused variable 'local' (unused-variable)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1855:4: C0103: Argument name "tz" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1868:23: W0212: Access to a protected member _local_timezone of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1892:4: W0221: Parameters differ from overridden 'isoformat' method (arguments-differ)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1907:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1912:8: C0103: Variable name "tz" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1914:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1920:8: C0103: Variable name "L" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1926:8: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1931:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1934:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1942:35: W0622: Redefining built-in 'format' (redefined-builtin)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1944:8: C0415: Import outside toplevel (_strptime) (import-outside-toplevel)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1945:15: W0212: Access to a protected member _strptime_datetime of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1987:8: R1705: Unnecessary "elif" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1995:8: R1705: Unnecessary "elif" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1994:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2003:8: R1705: Unnecessary "elif" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2002:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2011:8: R1705: Unnecessary "elif" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2010:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2019:8: R1705: Unnecessary "elif" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2018:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2026:4: W0221: Parameters differ from overridden '_cmp' method (arguments-differ)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2029:15: W0212: Access to a protected member _tzinfo of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2049:25: W0212: Access to a protected member _year of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2049:38: W0212: Access to a protected member _month of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2049:52: W0212: Access to a protected member _day of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2050:25: W0212: Access to a protected member _hour of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2050:38: W0212: Access to a protected member _minute of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2050:53: W0212: Access to a protected member _second of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2051:25: W0212: Access to a protected member _microsecond of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2053:12: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2061:8: R1706: Consider using ternary (1 if diff else 0) (consider-using-ternary)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2111:16: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2113:16: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2125:4: W0221: Parameters differ from overridden '_getstate' method (arguments-differ)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2129:8: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2131:12: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2135:8: R1705: Unnecessary "else" after "return" (no-else-return)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2140:33: W0621: Redefining name 'tzinfo' from outer scope (line 1141) (redefined-outer-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2143:19: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:1558:0: R0904: Too many public methods (26/20) (too-many-public-methods)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2170:4: C0103: Variable name "THURSDAY" doesn't conform to snake_case naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2179:0: C0103: Class name "timezone" doesn't conform to PascalCase naming style (invalid-name)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2202:8: W0212: Access to a protected member _offset of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2203:8: W0212: Access to a protected member _name of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2295:15: W0212: Access to a protected member _create of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2299:15: W0212: Access to a protected member _create of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2300:15: W0212: Access to a protected member _create of a client class (protected-access)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2500:4: W0401: Wildcard import _datetime (wildcard-import)</pre>
<pre>/log/firmware/firmware/usr/lib/python3.8/datetime.py:2500:4: W0614: Unused import datetime_CAPI from wildcard import (unused-wildcard-import)</pre>
<br />
<pre>-----------------------------------</pre>
<pre>Your code has been rated at 7.56/10</pre>
<br />
  <!-- content end -->
</div>

<a href="https://github.com/e-m-b-a/emba" title="github.com/e-m-b-a/emba" target="_blank">
  <div id="buttonInfo" class="rectButtonVisible">
    <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" width="40" height="40" style="-ms-transform: rotate(360deg); -webkit-transform: rotate(360deg); transform: rotate(360deg);" preserveAspectRatio="xMidYMid meet" viewBox="0 0 21 21"><g fill="none" fill-rule="evenodd" transform="translate(2 2)"><g stroke="white" stroke-linecap="round" stroke-linejoin="round"><circle cx="8.5" cy="8.5" r="8"/><path d="M8.5 12.5v-4h-1"/><path d="M7.5 12.5h2"/></g><circle cx="8.5" cy="5.5" fill="white" r="1"/></g></svg>
  </div>
</a>
<div id="buttonBack" class="nonClickable rectButtonVisible"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" width="40" height="40" style="-ms-transform: rotate(360deg); -webkit-transform: rotate(360deg); transform: rotate(360deg);" preserveAspectRatio="xMidYMid meet" viewBox="0 0 21 21"><path d="M9.5 14.5l4-4l-4-4" fill="none" stroke="#444" stroke-linecap="round" stroke-linejoin="round"/></svg></div>
<div id="buttonForward" class="nonClickable rectButtonVisible"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" width="40" height="40" style="-ms-transform: rotate(360deg); -webkit-transform: rotate(360deg); transform: rotate(360deg);" preserveAspectRatio="xMidYMid meet" viewBox="0 0 21 21"><path d="M11.5 14.5l-4-4l4-4" fill="none" stroke="#444" stroke-linecap="round" stroke-linejoin="round"/></svg></div> 
<div id="buttonDown" class="rectButton" onclick="scrollDown()"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" width="40" height="40" style="-ms-transform: rotate(360deg); -webkit-transform: rotate(360deg); transform: rotate(360deg);" preserveAspectRatio="xMidYMid meet" viewBox="0 0 21 21"><g fill="none" fill-rule="evenodd" stroke="#fff" stroke-linecap="round" stroke-linejoin="round"><path d="M14.5 6.5l-4 4l-4-4"/><path d="M14.5 10.5l-4 4l-4-4"/></g></svg></div>
<div id="buttonUp" class="rectButton" onclick="scrollUp()"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" width="40" height="40" style="-ms-transform: rotate(360deg); -webkit-transform: rotate(360deg); transform: rotate(360deg);" preserveAspectRatio="xMidYMid meet" viewBox="0 0 21 21"><g fill="none" fill-rule="evenodd" stroke="#fff" stroke-linecap="round" stroke-linejoin="round"><path d="M6.5 14.5l4-4l4 4"/><path d="M6.5 10.5l4-4l4 4"/></g></svg></div> 

<script>
window.onload = function () {
  var buttons = document.getElementsByClassName("rectButton");
  for (let i=0; i<buttons.length; i++) {
    buttons[i].style.visibility = "visible";
  }
  var buttonDown = document.getElementById("buttonDown");
  buttonDown.classList.add('rectButtonVisible');
  buttonDown.classList.remove('rectButton');

  var buttonUp = document.getElementById("buttonUp");
  buttonUp.classList.add('rectButtonVisible');
  buttonUp.classList.remove('rectButton');
    
  var buttonSizer = document.getElementById("buttonSizer");
  buttonSizer.innerHTML = '<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" width="40" height="40" style="-ms-transform: rotate(360deg); -webkit-transform: rotate(360deg); transform: rotate(360deg);" preserveAspectRatio="xMidYMid meet" viewBox="0 0 21 21"><g fill="none" fill-rule="evenodd" stroke="white" stroke-linecap="round" stroke-linejoin="round"><path d="M9.5 16.5v-4.978l-5-.022"/><path d="M18.5 2.5l-7 7"/><path d="M16.5 9.5l-5 .023V4.5"/><path d="M9.5 11.5l-7 7"/></g></svg>';
  if (document.getElementById("buttonTime") !== null) {
    var buttonTime = document.getElementById("buttonTime");
    buttonTime.style.bottom = "50px";
  }
  document.getElementById("expand").value = "+ Show more results";
  document.getElementById("expand").className += " expand";
}

function openNav() {
  document.getElementById("sidenav").style.width = "250px";
  document.getElementById("main").style.marginLeft = "250px";
  document.getElementById("nav").style.visibility = "visible";
}

function closeNav() {
  document.getElementById("sidenav").style.width = "50px";
  document.getElementById("sidenav").scrollTop = 0;
  document.getElementById("main").style.marginLeft = "50px";
  document.getElementById("nav").style.visibility = "hidden";
}

function scrollDown() {
  window.scrollTo(0,document.body.scrollHeight);
}

function scrollUp() {
  window.scrollTo(0,0);
}

function buttonNav() {
  var button = document.getElementById("buttonSizer");
  if ( button.classList.contains('buttonOpen') ) {
    button.classList.remove('buttonOpen');
    button.classList.add('buttonClose');
    button.innerHTML = '<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" width="40" height="40" style="-ms-transform: rotate(360deg); -webkit-transform: rotate(360deg); transform: rotate(360deg);" preserveAspectRatio="xMidYMid meet" viewBox="0 0 21 21"><g fill="none" fill-rule="evenodd" stroke="white" stroke-linecap="round" stroke-linejoin="round"><path d="M9.5 16.5v-4.978l-5-.022"/><path d="M18.5 2.5l-7 7"/><path d="M16.5 9.5l-5 .023V4.5"/><path d="M9.5 11.5l-7 7"/></g></svg>';
    openNav();
  } else if( button.classList.contains('buttonClose') ) {
    button.classList.remove('buttonClose');
    button.classList.add('buttonOpen');
    button.innerHTML = '<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-hidden="true" focusable="false" width="40" height="40" style="-ms-transform: rotate(360deg); -webkit-transform: rotate(360deg); transform: rotate(360deg);" preserveAspectRatio="xMidYMid meet" viewBox="0 0 21 21"><g fill="none" fill-rule="evenodd" stroke="white" stroke-linecap="round" stroke-linejoin="round"><path d="M18.5 7.5v-5h-5"/><path d="M18.5 2.5l-6 5.929"/><path d="M7.5 18.5l-5 .023V13.5"/><path d="M8.5 12.5l-6 6"/></g></svg>';
    closeNav();
  };
}

document.onkeydown = function(e) {
    switch (e.keyCode) {
      case 37:
        document.getElementById("buttonForward").parentNode.click(); 
        break;
      case 39:
        document.getElementById("buttonBack").parentNode.click();       
        break;
    }
};
</script>

</body>
</html>
